### ISSUE TITLE
knife need to download/upload nodes without ohai data

### ISSUE THREAD

**Issue by llibicpep** [chef/chef#7373]
## Description

This issue has been there since 2014 https://tickets.opscode.com/browse/CHEF-5151 but seems didn't made it through migration to github. I do have slightly different justification for that issue, though. Which I hope will slightly raise a priority for it.

So let's say I do a GitOps, and it perfectly applies to roles, environments and data bags. If I were to change some of it, I'll use knife with `-z` flag, submit a PR with my change to my git repository, do a standard code review process and some CI checks, merge it and then my CI would pick it up and do `knife upload` to it. For a cattle-style infrastructure that is all you need, really. Everything is too versatile and nodes goes up and down and you don't want to store anything node-level.

But sometimes you still might have a pet-style pieces in your infrastructure too. And all of the sudden you can't easily apply that pattern there, as knife download/upload commands will include ohai data too. This still would be achievable, didn't tried it yet, but something like do a loop through `knife node list` with `knife node show -F json <it>` to dump initial data and keep it in sync with all new provisions and decommissions, and use `knife node from file` or something for changes to node env or runlist or attributes, but the only reason I still didn't made it because I'm lazy and would love to see it possible same way as I already do it for roles and envs and data bags.

TL;DR: Please enable GitOps style PR driven changes for node objects same way it is possible today for roles, environments and data bags!

## Chef Version
N/A

## Platform Version
N/A

## Replication Case
Do a GitOps

## Client Output
N/A

## Stacktrace
N/A
